cdm/core/src/main/java/ucar/nc2/dataset/VariableDS.java

```diff
@@ -609,172 +613,211 @@ public void removeCoordinateSystem(CoordinateSystem cs) {
 
   @Override
   public boolean hasScaleOffset() {
-    return scaleMissingUnsignedProxy.hasScaleOffset();
+    return scaleOffset != null;
   }
 
   @Override
   public double getScaleFactor() {
-    return scaleMissingUnsignedProxy.getScaleFactor();
+    return scaleOffset != null ? scaleOffset.getScaleFactor() : 1;
   }
 
   @Override
   public double getOffset() {
-    return scaleMissingUnsignedProxy.getOffset();
+    return scaleOffset != null ? scaleOffset.getOffset() : 0;
   }
 
   @Override
   public boolean hasMissing() {
-    return scaleMissingUnsignedProxy.hasMissing();
+    return convertMissing != null && convertMissing.hasMissing();
   }
 
   @Override
   public boolean isMissing(double val) {
-    return scaleMissingUnsignedProxy.isMissing(val);
+    if (Double.isNaN(val)) {
+      return true;
+    }
+    return convertMissing != null && convertMissing.isMissing(val);
   }
 
   @Override
   public boolean hasValidData() {
-    return scaleMissingUnsignedProxy.hasValidData();
+    return convertMissing != null && convertMissing.hasMissingValue();
   }
 
   @Override
   public double getValidMin() {
-    return scaleMissingUnsignedProxy.getValidMin();
+    return convertMissing != null ? convertMissing.getValidMin() : -Double.MAX_VALUE;
   }
 
   @Override
   public double getValidMax() {
-    return scaleMissingUnsignedProxy.getValidMax();
+    return convertMissing != null ? convertMissing.getValidMax() : Double.MAX_VALUE;
   }
 
   @Override
   public boolean isInvalidData(double val) {
-    return scaleMissingUnsignedProxy.isInvalidData(val);
+    return convertMissing != null && convertMissing.isInvalidData(val);
   }
 
   @Override
   public boolean hasFillValue() {
-    return scaleMissingUnsignedProxy.hasFillValue();
+    return hasFillValue;
   }
 
   @Override
   public double getFillValue() {
-    return scaleMissingUnsignedProxy.getFillValue();
+    return fillValue;
   }
 
   @Override
   public boolean isFillValue(double val) {
-    return scaleMissingUnsignedProxy.isFillValue(val);
+    return convertMissing != null && convertMissing.isFillValue(val);
   }
 
   @Override
   public boolean hasMissingValue() {
-    return scaleMissingUnsignedProxy.hasMissingValue();
+    return convertMissing != null && convertMissing.hasMissingValue();
   }
 
   @Override
   public double[] getMissingValues() {
-    return scaleMissingUnsignedProxy.getMissingValues();
+    return convertMissing != null ? convertMissing.getMissingValues() : new double[] {0};
```

Comment:
if `convertMissing == null` shouldn't this be an empty array like
```suggestion
    return convertMissing != null ? convertMissing.getMissingValues() : new double[0];
```